### PR TITLE
refactor: replace request with axios

### LINKED ISSUE #1733: 2.88.2 Request has been deprecated
I am facing issue while installing expo cli .
after installing node and git . i am running the command " npm install expo-cli --global"
but getting issue and installation stops at 
"npm WARN deprecated request@2.88.2: request has been deprecated, see https://github.com/request/request/issues/3142".
Anyone please tell me why am i facing this issue? Looking for positive and helpful response for you .
Thanks

### PR DESCRIPTION
**Pull request by byCedric**
This fixes #1733, made together with @fson ❤️

### CODE REVIEW

File: packages/expo-cli/src/commands/build/ios/utils/image.ts [hunk 1]
```diff
@@ -1,18 +1,30 @@
...
 import pick from 'lodash/pick';
 import { XDLError } from '@expo/xdl';
-import request from 'request';
+import axios from 'axios';
 import validator from 'validator';
 
-export async function ensurePNGIsNotTransparent(imagePathOrURL: string): Promise<void> {
-  let hasAlreadyResolved = false;
-  const stream = validator.isURL(imagePathOrURL, {
+async function getImageStreamAsync(imagePathOrURL: string) {
```
Comment (EvanBacon) [inline packages/expo-cli/src/commands/build/ios/utils/image.ts:10]: We should try to combine this with [image-utils](https://github.com/expo/expo-cli/blob/89ad047fb096c6a17ada65380c7c0fed822135ca/packages/image-utils/src/Download.ts)

**Comment by brentvatne** [comment]
can you guys explain how this was tested? it impacts turtle so we need to be careful here

**Comment by byCedric** [comment]
We've used Axios here because it's already used in e.g. [APIv2](https://github.com/expo/expo-cli/blob/018dbc427426036539210d1108a25e2049821590/packages/xdl/src/ApiV2.ts) and the [XDL](https://github.com/expo/expo-cli/search?q=axios&unscoped_q=axios). So my understanding was we already chose for Axios. If this is not the case, I would be happy to discuss alternatives 😄

**Comment by brentvatne** [comment]
@dsokal - heads up that this code impacts turtle

**Comment by dsokal** [comment]
Thanks for the heads-up Brent. 
All of these changes seem safe. I'll test them with Turtle next time I bump xdl there.

**Pull request merged**